subsys/spm/spm.c

```diff
@@ -328,12 +328,16 @@ static void spm_config_peripherals(void)
 		PERIPH("NRF_FPU", NRF_FPU_S, CONFIG_SPM_NRF_FPU_NS),
 		PERIPH("NRF_EGU1", NRF_EGU1_S, CONFIG_SPM_NRF_EGU1_NS),
 		PERIPH("NRF_EGU2", NRF_EGU2_S, CONFIG_SPM_NRF_EGU2_NS),
-		PERIPH("NRF_TWIM2", NRF_TWIM2_S, CONFIG_SPM_NRF_TWIM2_NS),
-		PERIPH("NRF_SPIM3", NRF_SPIM3_S, CONFIG_SPM_NRF_SPIM3_NS),
-		PERIPH("NRF_TIMER0", NRF_TIMER0_S, CONFIG_SPM_NRF_TIMER0_NS),
-		PERIPH("NRF_TIMER1", NRF_TIMER1_S, CONFIG_SPM_NRF_TIMER1_NS),
-		PERIPH("NRF_TIMER2", NRF_TIMER2_S, CONFIG_SPM_NRF_TIMER2_NS),
-		PERIPH("NRF_SAADC", NRF_SAADC_S, CONFIG_SPM_NRF_SAADC_NS),
+		PERIPH("NRF_TWIM2", NRF_TWIM2, CONFIG_SPM_NRF_TWIM2_NS),
+		PERIPH("NRF_SPIM3", NRF_SPIM3, CONFIG_SPM_NRF_SPIM3_NS),
+		PERIPH("NRF_TIMER0", NRF_TIMER0, CONFIG_SPM_NRF_TIMER0_NS),
+		PERIPH("NRF_TIMER1", NRF_TIMER1, CONFIG_SPM_NRF_TIMER1_NS),
+		PERIPH("NRF_TIMER2", NRF_TIMER2, CONFIG_SPM_NRF_TIMER2_NS),
+		PERIPH("NRF_SAADC", NRF_SAADC, CONFIG_SPM_NRF_SAADC_NS),
+		PERIPH("NRF_PWM0", NRF_PWM0, CONFIG_SPM_NRF_PWM0_NS),
+		PERIPH("NRF_PWM1", NRF_PWM1_S, CONFIG_SPM_NRF_PWM1_NS),
+		PERIPH("NRF_PWM2", NRF_PWM2_S, CONFIG_SPM_NRF_PWM2_NS),
```

Comment:
Like this it looks really messy; PWM0 exists but PWM1,2,3 require the _S.
I'd like a clean solution, here, i.e. define the PWMs as "ok" and use the standard macros

Reply:
I agree it doesn't look good, but defining the instances "ok" is an upstream fix as far as I can see.

Reply:
@hakonfam  @jtguggedal  Is there any dts overlay for nrf9160_pca10090 when we build the SPM? If so, these could be defined there as "ok". The justification would be that "the SPM needs them to set them to non-secure" :)

Reply:
If it's possible to pass an overlay to the SPM build, I agree that's a better solution. Do you know if that already happens or is possible, @hakonfam ?

Reply:
Looks like it's the `SPM` sample that's built, so I'll add the peripherals in the already existing overlay there.